common/djangoapps/entitlements/tests/test_entitlements_api.py

```diff
@@ -0,0 +1,162 @@
+import json
+import unittest
+import uuid
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from entitlements.tests.factories import CourseEntitlementFactory
+from entitlements.models import CourseEntitlement
+from student.tests.factories import CourseEnrollmentFactory, UserFactory, TEST_PASSWORD
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class EntitlementsTest(ModuleStoreTestCase):
+    """
+    Entitlements API/View Tests
+    """
+    USERNAME = 'Bob'
+    ENABLED_CACHES = ['default']
+    COURSE_RUN_ID = 'some/great/course'
+
+    def setUp(self):
+        super(EntitlementsTest, self).setUp()
+        self.user = UserFactory(is_staff=True)
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.course = CourseFactory.create(org='TestX', course='TS101', run='T1')
+        # self.entitlements_url = '/api/entitlements/v1/entitlements/'
+        self.entitlements_url = reverse('entitlements_api:entitlements-list')
+        self.entitlements_uuid_path = 'entitlements_api:api:entitlements-list'
+        self.course_uuid = str(uuid.uuid4())
+
+    def _get_data_set(self, user):
+        """
+        Get a basic data set for an entitlement
+        """
+        return {
+            "user": user.username,
+            "mode": "verified",
+            "course_uuid": self.course_uuid,
+            "order_number": "EDX-1001"
+        }
+
+    def test_auth_required(self):
+        self.client.logout()
+        response = self.client.get(self.entitlements_url)
+        assert response.status_code == 401
+
+    def test_staff_user_required(self):
+        not_staff_user = UserFactory()
+        self.client.login(username=not_staff_user.username, password=UserFactory._DEFAULT_PASSWORD)
+        response = self.client.get(self.entitlements_url)
+        assert response.status_code == 403
+
+    def test_add_entitlement_with_missing_data(self):
+        entitlement_data_missing_parts = self._get_data_set(self.user)
+        entitlement_data_missing_parts.pop('mode')
+        entitlement_data_missing_parts.pop('course_uuid')
+
+        response = self.client.post(
+            self.entitlements_url,
+            data=json.dumps(entitlement_data_missing_parts),
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+
+    def test_add_entitlement(self):
+        entitlement_data = self._get_data_set(self.user)
+
+        response = self.client.post(
+            self.entitlements_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 201
+        results = response.data
```

Comment:
Given that you only use this variable once, don't bother making the variable: `assert response.data is not None`